code/modules/projectiles/ammo_datums.dm

```diff
@@ -2468,10 +2468,96 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 
 /datum/ammo/energy/lasgun/marine/heavy_laser/do_at_max_range(turf/T, obj/projectile/P)
-	drop_nade(T.density ? P.loc : T)
+	drop_nade(T.density ? get_step_towards(T, P) : T)
+
+//psyblast// lots of placeholder stuff
+
+/datum/ammo/energy/xeno
+	///This text will show up when this ammo type is selected by a xeno. Span proc should be applied when this var is used.
+	var/select_text
+	///Plasma cost to fire this projectile
+	var/plasma_cost
+
+/datum/ammo/energy/xeno/psy_blast
+	name = "psychic blast"
+	flags_ammo_behavior = AMMO_XENO|AMMO_EXPLOSIVE|AMMO_ROCKET|AMMO_ENERGY|AMMO_SUNDERING|AMMO_HITSCAN
+	damage = 35
+	penetration = 10
+	sundering = 1
+	max_range = 30
+	accurate_range = 15
+	hitscan_effect_icon = "beam_cult"
+	icon_state = "psy_blast"
+	select_text = "We will now fire a psychic blast. These have an area of effect knockback when struck."
+	plasma_cost = 230
+	///The AOE for drop_nade
+	var/aoe_range = 3
+
+/datum/ammo/energy/xeno/psy_blast/drop_nade(turf/T, obj/projectile/P)
+	if(!T || !isturf(T))
+		return
+	playsound(T, 'sound/effects/EMPulse.ogg', 50)
+	for(var/atom/movable/victim in view(aoe_range, T))
+		if(victim.anchored)
+			continue
+		if(isliving(victim))
+			var/mob/living/living_victim = victim
+			if(living_victim.stat == DEAD)
+				continue
+			if(!isxeno(living_victim))
+				living_victim.apply_damage(35, BURN, blocked = living_victim.get_soft_armor(ENERGY), updating_health = TRUE)
+				staggerstun(living_victim, P, 9, slowdown = 1)
+		else
+			victim.ex_act(EXPLODE_LIGHT)
+		var/throw_dir = get_dir(T, victim)
+		if(T == get_turf(victim))
+			throw_dir = get_dir(P.starting_turf, T)
+		victim.safe_throw_at(get_ranged_target_turf(T, throw_dir, 5), 4, 1, spin = TRUE)
+	new /obj/effect/temp_visual/shockwave(T, aoe_range + 2)
+
+/datum/ammo/energy/xeno/psy_blast/on_hit_mob(mob/M, obj/projectile/P)
+	drop_nade(get_turf(M), P)
+
+/datum/ammo/energy/xeno/psy_blast/on_hit_obj(obj/O, obj/projectile/P)
+	drop_nade(get_turf(O), P)
+
+/datum/ammo/energy/xeno/psy_blast/on_hit_turf(turf/T, obj/projectile/P)
+	drop_nade(T.density ? get_step_towards(T, P) : T, P)
+
+/datum/ammo/energy/xeno/psy_blast/do_at_max_range(turf/T, obj/projectile/P)
+	drop_nade(T.density ? get_step_towards(T, P) : T, P)
+
+/datum/ammo/energy/xeno/psy_blast/psy_lance
+	name = "psychic lance"
+	flags_ammo_behavior = AMMO_XENO|AMMO_ENERGY|AMMO_SUNDERING|AMMO_HITSCAN|AMMO_PASS_THROUGH_MOVABLE
+	damage = 60
+	penetration = 50
+	accuracy = 100
+	sundering = 5
+	max_range = 12
+	hitscan_effect_icon = "beam_hcult"
+	icon_state = "psy_lance"
+	select_text = "We will now fire a psychic lance. These are powerful piercing beams, particularly strong against armored targets."
+	plasma_cost = 300
+
+/datum/ammo/energy/xeno/psy_blast/psy_lance/on_hit_obj(obj/O, obj/projectile/P)
+	if(ismecha(O))
+		var/obj/vehicle/sealed/mecha/mech_victim = O
+		mech_victim.take_damage(rand(200, 400), BURN, ENERGY, 0, armour_penetration = penetration)
+	else
+		O.ex_act(EXPLODE_HEAVY)
```

Comment:
this should be using acid armor and not using ex act as bomb is for FF and energy is for HvH

Reply:
why acid armour, and why not energy? Mimir provides large acid protection, while this is a literal energy beam. I'd be fine with laser I guess, but I wouldn't want it to be acid ideally.